server/src/handler_driver.c

```diff
@@ -217,7 +669,9 @@ static bool drive_route_progressive_stop_signals(const char *train_id, t_interlo
 static bool drive_route(const int grab_id, const char *route_id, const bool is_automatic) {
 	char *train_id = strdup(grabbed_trains[grab_id].name->str);
 	t_interlocking_route *route = get_route(route_id);
-	if (!drive_route_params_valid(train_id, route)) {
+	if (train_id == NULL || route == NULL || !drive_route_params_valid(train_id, route)) {
+		syslog_server(LOG_WARNING, "Drive route: Invalid params at start");
```

Comment:
True. One more thing I noticed here: It would perhaps be useful to have a way to relate this back to a particular request/train/route. Of course, if the train and/or route are invalid as in == NULL, this is difficult. Still, for debugging multiple route requests, I'd want to know which one failed.    
This also (distantly) concerns the message that is returned to the user upon a failure - if drive_route returns false, i.e. fails, handler_drive_route just returns the OCS_NOT_IMPLEMENTED code with no other information.

Reply:
A monotonically increasing `route_request_id` could be passed through all the function calls.